common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -470,9 +494,69 @@ private SwitchBrowserProtocolCoordinator getSwitchBrowserCoordinator() {
 
     /**
      * Set the switch browser bundle to be used when resuming the flow.
+     *
      * @param bundle The bundle containing the data needed to resume the flow.
      */
     public static synchronized void setSwitchBrowserBundle(@Nullable final Bundle bundle) {
         switchBrowserBundle = bundle;
     }
+
+
+    /**
+     * Sets up the PasskeyWebListener if the request headers indicate that both authentication and registration
+     * are supported. If the hook fails, it downgrades to authentication only.
+     * Called during WebView setup.
+     */
+    private void setupPasskeyWebListener(@NonNull final WebView webView,
+                                         @NonNull final AzureActiveDirectoryWebViewClient webViewClient) {
+        final String methodTag = TAG + ":setupPasskeyWebListener";
+        final String passkeyProtocolHeader = mRequestHeaders.get(FidoConstants.PASSKEY_PROTOCOL_HEADER_NAME);
+        if (FidoConstants.PASSKEY_PROTOCOL_HEADER_AUTH_AND_REG.equals(passkeyProtocolHeader)) {
+            final boolean passkeyWebListenerHooked = PasskeyWebListener.hook(webView, requireActivity(), webViewClient);
+            if (!passkeyWebListenerHooked) {
+                Logger.warn(methodTag, "PasskeyWebListener hook failed, Downgrading to auth only.");
+                // Downgrade to auth only
+                mRequestHeaders.put(FidoConstants.PASSKEY_PROTOCOL_HEADER_NAME, FidoConstants.PASSKEY_PROTOCOL_HEADER_AUTH_ONLY);
+            }
+        } else {
+            Logger.warn(methodTag, "Passkey protocol header not found or not for both auth and reg." +
+                    " Not hooking the PasskeyWebListener.");
+        }
+    }
+
+    /**
+     * Injects the Passkey protocol header into the request headers if the WebAuthN query parameter is present.
+     * If the header already exists, it will not be modified. If the request is from broker and the Passkey registration flight is enabled,
+     * the header will indicate support for both authentication and registration.
+     *
+     * @param requestHeaders The request headers to modify.
+     */
+    private void injectPasskeyProtocolHeader(@NonNull final HashMap<String, String> requestHeaders) {
+        final String methodTag = TAG + ":injectPasskeyProtocolHeader";
+        final Uri authRequestUri = Uri.parse(mAuthorizationRequestUrl);
+        final String webAuthNQueryParameter = authRequestUri.getQueryParameter(FidoConstants.WEBAUTHN_QUERY_PARAMETER_FIELD);
+        final boolean hasWebAuthNQueryParameter = !StringUtil.isNullOrEmpty(webAuthNQueryParameter);
+
+        if (!hasWebAuthNQueryParameter) {
```

Comment:
nit: no need for the hasWebAuthNQueryParameter variable if it's only going to be used for this condition